src/execute.c

```diff
@@ -348,3 +348,21 @@ print_command(FILE *fp, char **argv)
 	}
 	fprintf(fp, "\n");
 }
+
+char *
+string_command(char **argv)
+{
+	size_t len = 0;
+	for (int i = 0; argv[i]; i++) {
+		len += (i == 0) ? 0 : 1;
+		len += strlen(argv[i]);
+	}
+	len += 1;
+	char *buf = x_calloc(1, len + 1);
+	for (int i = 0; argv[i]; i++) {
+		strcat(buf, (i == 0) ? "" : " ");
```

Comment:
For very long command lines, this O(n²) algorithm might hurt performance... It would be nice to get away with the `strcat`s and keep a pointer to remember where to copy the next argument instead.

Reply:
Hmm, strcat should be O(n) and not O(n²) ? It is supposed to keep going until it finds NUL.

Reply:
Of course, that would be `strcpy` and not `strcat` - you are right, will change them as suggested